### PR TITLE
Allow multiple docker engines/mesos slaves in the mesos tests

### PR DESCRIPTION
**Pull request by vieux**
:balloon: :balloon: :balloon: :balloon: :balloon:

### CODE REVIEW

**Comment by vieux** [comment]
ping @aluzzardi @abronan

**Comment by vieux** [comment]
ping @tnachen

**Comment by jimenez** [comment]
LGTM

**Comment by abronan** [comment]
LGTM :facepunch:

File: test/integration/mesos/mesos_helpers.bash [hunk 1]
```diff
@@ -4,38 +4,39 @@ load ../helpers
...
-	MESOS_SLAVE=$(
-	docker_host run --privileged -d --name mesos-slave --volumes-from node-0 -e DOCKER_HOST="${HOSTS[0]}" -v /sys/fs/cgroup:/sys/fs/cgroup --net=host \
-	$MESOS_IMAGE /mesos/build/bin/mesos-slave.sh --master=127.0.0.1:$MESOS_MASTER_PORT --containerizers=docker --hostname=127.0.0.1 --port=$(($MESOS_MASTER_PORT + 1))
-	       )
-
-	retry 10 1 eval "docker_host ps | grep 'mesos-slave'"
+	for ((i=0; i < current; i++)); do
+	    MESOS_SLAVES[$i]=$(
+		docker_host run --privileged -d --name mesos-slave-$i --volumes-from node-$i -e DOCKER_HOST="${HOSTS[$i]}" -v /sys/fs/cgroup:/sys/fs/cgroup --net=host \
+		$MESOS_IMAGE /mesos/build/bin/mesos-slave.sh --master=127.0.0.1:$MESOS_MASTER_PORT --containerizers=docker --attributes="docker_port:${PORTS[$i]}" --hostname=127.0.0.1 --port=$(($MESOS_MASTER_PORT + (1 + $i)))
```
Comment (aluzzardi) [inline test/integration/mesos/mesos_helpers.bash]: Maybe maintaining the `PORTS` array is not required.

You could do something like:
```
local port=$(echo ${HOSTS[$i]} | cut -d: -f1)
```

to split the port from `HOSTS`

Comment (aluzzardi) [inline test/integration/mesos/mesos_helpers.bash]: Actually - what's the difference between this port:
```
--attributes="docker_port:${PORTS[$i]}"
```

And this one:
```
 --port=$(($MESOS_MASTER_PORT + (1 + $i)))
```

**Comment by aluzzardi** [comment]
Minor comments

Otherwise, LGTM

File: test/integration/mesos/mesos_helpers.bash (hunk 1 above)
Comment (vieux) [inline test/integration/mesos/mesos_helpers.bash]: `--attributes=docker_port` is a custom data to apply to the slave (kind of daemon labels) the default would be 2375
`--port` is the port the master will bind on. the default would be 5050

Comment (vieux) [inline test/integration/mesos/mesos_helpers.bash]: I'l try to remove `HOSTS`

**Comment by tnachen** [comment]
@vieux this should be ready right?

**Comment by vieux** [comment]
@tnachen yes, it was depending on another PR that I just merged.

**Pull request merged**